src/led.rs

```diff
@@ -10,69 +11,87 @@ mod command_nr {
     pub const TOGGLE: usize = 3;
 }
 
-pub struct Led {
-    led_num: usize,
+pub struct LedDriver {
+    pub(crate) _unconstructible: (),
 }
 
-pub fn count() -> TockResult<usize> {
-    command(DRIVER_NUMBER, command_nr::COUNT, 0, 0).map_err(Into::into)
+pub struct Led<'a> {
+    led_num: usize,
+    phantom: PhantomData<&'a mut ()>,
 }
 
-pub fn get(led_num: usize) -> Option<Led> {
-    if led_num < count().ok().unwrap() {
-        Some(Led { led_num })
-    } else {
-        None
+impl LedDriver {
+    pub fn get(&mut self, led_num: usize) -> Option<Led> {
```

Comment:
Same here. Please don't change the API too much. Migrating to the singleton is okay but I would be happy if changing/adding methods were avoided.

Reply:
I agree. Making `get` consume  `&self` shouldn't violate the safety guarantees of rust.

Reply:
fixed.